src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Linux/LinuxUtilizationParserCgroupV2.cs

```diff
@@ -555,16 +558,18 @@ private static bool TryGetCgroupRequestCpu(IFileSystem fileSystem, out float cpu
             Throw.InvalidOperationException($"Could not parse '{_cpuPodWeight}'. Expected to get an integer but got: '{cpuPodWeight}'.");
         }
 
+        if (cpuPodWeight < CpuPodWeightPossibleMin || cpuPodWeight > CpuPodWeightPossibleMax)
+        {
+            Throw.ArgumentOutOfRangeException($"Expected to see cpuPodWeight='{cpuPodWeight}' in range [{CpuPodWeightPossibleMin}-{CpuPodWeightPossibleMax}]");
+        }
+
         // Calculate CPU pod request in millicores based on the weight, using the formula:
         // y = (1 + ((x - 2) * 9999) / 262142), where y is the CPU weight and x is the CPU share (cgroup v1)
         // https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
-        long cpuPodShare = ((cpuPodWeight * 262142) + 19997) / 9999;
-        if (cpuPodShare == -1)
-        {
-            Throw.InvalidOperationException($"Could not calculate CPU share from CPU weight '{cpuPodShare}'");
-        }
+#pragma warning disable S109 // Magic numbers should not be used - using the formula, forgive.
+        cpuUnits = ((cpuPodWeight - 1) * 262142 / 9999) + 2;
```

Comment:
[nit] The folrmula and the comment on line:567 and line:570 do not quite match.
The formulas are invereted - i.e., l:567 => y = ..., and l: 570: x = ....

It would be great to update the comment to add a missing link between x and y and cpuUnits  and cpuPodWeight.
E.g., something like:
```
y = (1 + ((x - 2) * 9999) / 262142), where y is the CPU weight (i.e., cpuPodWeight) and x is the CPU share (cpuUnits).
```

Maybe update the formula too.

Reply:
Updated